source/nsdynmemLIB/nsdynmemLIB.c

```diff
@@ -35,6 +36,26 @@ typedef enum mem_stat_update_t {
 
 static mem_stat_t *mem_stat_info_ptr = 0;
 
+typedef struct {
+    ns_list_link_t link;
+} hole_t;
+
+typedef NS_LIST_HEAD(hole_t, link) holes_list_t;
```

Comment:
Don't think you need the typedef, so just NS_LIST_DEFINE would do. (That also initialises to empty, not that you need it).